### PR TITLE
fix: [#2320] check for correct betrokkene rol when accessing single zaak

### LINKED ISSUE #2320: Role content should be checked in case access mixin if limit_user_visible_cases_to_role is set
## Summary
Gebruiker kan zaken vinden waar hij/zij betrokkene van is doormiddel van direct searches in de URL balk. Deze zaken zouden eigenlijk niet zichtbaar mogen zijn door de instelling `config.limit_user_visible_cases_to_role`.

### Test case settings/context: 
1. `config.limit_user_visible_cases_to_role = 'initiator'`
2. gebruiker is betrokkene van een zaak.

### PR DESCRIPTION
**Pull request by pi-sigma**
## Summary

The `CaseAccessMixin` only checked if the user had *some* role for the zaak they're trying to access, not if they have the *correct* role (e.g. the role 'initiator' when visibility is restricted to the initiator of a case via `OpenZaakConfig`).

## Issue References

Fixes #2320

## Checklist

- [x] CHANGELOG.rst updated

### CODE REVIEW

File: src/open_inwoner/cms/cases/views/mixins.py [hunk 1]
```diff
@@ -132,38 +132,59 @@ def dispatch(self, request, *args, **kwargs):
             client = api_group.zaken_client
             self.zaak = client.fetch_single_zaak(object_id)
             if self.zaak:
-                # check if we have a role in this case
+                # check if we have the correct role in this case
+                config = OpenZaakConfig.get_solo()
+                rol_restricted = config.limit_user_visible_cases_to_role
                 if request.user.bsn:
-                    if not client.fetch_roles_for_zaak_and_bsn(
+                    rolen = client.fetch_roles_for_zaak_and_bsn(
```
Comment (swrichards) [inline src/open_inwoner/cms/cases/views/mixins.py]:
```suggestion
                    rollen = client.fetch_roles_for_zaak_and_bsn(
```

(And elsewhere)

File: src/open_inwoner/cms/cases/views/mixins.py [hunk 2]
```diff
@@ -132,38 +132,59 @@ def dispatch(self, request, *args, **kwargs):
             client = api_group.zaken_client
             self.zaak = client.fetch_single_zaak(object_id)
             if self.zaak:
-                # check if we have a role in this case
+                # check if we have the correct role in this case
+                config = OpenZaakConfig.get_solo()
+                rol_restricted = config.limit_user_visible_cases_to_role
```
Comment (swrichards) [inline src/open_inwoner/cms/cases/views/mixins.py]: Let's be verbose:

```suggestion
                limit_access_to_role = config.limit_user_visible_cases_to_role
```

File: src/open_inwoner/openzaak/tests/test_case_detail.py [hunk 3]
```diff
@@ -2074,6 +2074,128 @@ def test_no_access_as_vestiging_when_no_roles_are_found_for_vestigingsnummer(
...
+    def test_no_access_when_incorrect_role_for_user_bsn(self, m):
+        """
+        Even if the user has a role for the case, access is denied when
+        `limit_user_visible_cases_to_role` is set and the role doesn't match.
+        """
+        behandelaar_role_for_user = generate_oas_component_cached(
+            "zrc",
+            "schemas/Rol",
+            url=f"{ZAKEN_ROOT}rollen/bb353aa-ad2c-4a07-ae75-15add5823",
+            omschrijvingGeneriek=RolOmschrijving.behandelaar,
```
Comment (swrichards) [inline src/open_inwoner/openzaak/tests/test_case_detail.py]: Could we do this in a subTest, enumerating over all values for `RolOmschrijving` (Except initiator, obviously)

File: src/open_inwoner/openzaak/tests/test_case_detail.py
Comment (swrichards) [inline src/open_inwoner/openzaak/tests/test_case_detail.py:1]: Can we also test the "no role" and "empty string role" cases?

**Review by swrichards — CHANGES_REQUESTED** [review]

File: src/open_inwoner/openzaak/tests/test_case_detail.py
Comment (pi-sigma) [reply src/open_inwoner/openzaak/tests/test_case_detail.py:1]: The "no role" case is already covered by the existing test suite, I've added the "empty string role" case to the new tests.

**Review by swrichards — APPROVED** [review]
